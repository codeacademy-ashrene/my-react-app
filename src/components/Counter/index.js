import React, { Component } from 'react';
import Button from '../Button'
class Counter extends Component {
  state = {
    counter: this.props.initialValue,
  }
  increment = () => {
    this.setState({counter: this.state.counter+1});
  }
  decrement = () => {
    this.setState({counter: this.state.counter-1});
  }
  render() {
    return (
      <div>
        <div>
          Counter: {this.state.counter}
        </div>
        <Button buttonText = "Increase" onClick = {this.increment}/>
        <Button buttonText = "Decrease" onClick = {this.decrement}/>
      </div>
    );
  }
}

export default Counter;
