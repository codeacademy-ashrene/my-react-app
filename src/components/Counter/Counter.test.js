import React from 'react';
import Counter from '.';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
describe('Counter', () => {
  it('should render counter', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should set initial value', () => {
    const wrapper = shallow(<Counter initialValue={10}/>);
    expect(wrapper.instance().state.counter).toEqual(10);
  })
  it('should set increment counter on increment', () => {
    const wrapper = shallow(<Counter initialValue={10}/>);
    wrapper.instance().increment();
    expect(wrapper.instance().state.counter).toEqual(11);
  })
  it('should set decrement counter on decrement', () => {
    const wrapper = shallow(<Counter initialValue={10}/>);
    wrapper.instance().decrement();
    expect(wrapper.instance().state.counter).toEqual(9);
  })
});