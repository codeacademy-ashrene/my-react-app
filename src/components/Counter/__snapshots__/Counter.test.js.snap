// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render counter 1`] = `
<div>
  <div>
    Counter: 
  </div>
  <div>
    <button
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "color": "#0000ff",
        }
      }
      type="button"
    >
       
      Increase
       
    </button>
  </div>
  <div>
    <button
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "color": "#0000ff",
        }
      }
      type="button"
    >
       
      Decrease
       
    </button>
  </div>
</div>
`;
